geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/utils/GeoJsonInference.scala

```diff
@@ -0,0 +1,292 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.tools.utils
+
+import java.io.InputStream
+import java.{lang => jl}
+import java.time.format.DateTimeParseException
+
+import com.vividsolutions.jts.geom._
+import com.typesafe.config.{Config, ConfigFactory, ConfigValueFactory}
+import com.fasterxml.jackson.core.{JsonFactory, JsonParser, JsonToken}
+import com.fasterxml.jackson.core.JsonToken._
+import org.geotools.feature.AttributeTypeBuilder
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder
+import org.opengis.feature.`type`.AttributeDescriptor
+import org.opengis.feature.simple.SimpleFeatureType
+import org.locationtech.geomesa.utils.io.PathUtils
+import org.locationtech.geomesa.utils.io.fs.FileSystemDelegate
+import org.locationtech.geomesa.utils.geotools.{AttributeSpec, CRS_EPSG_4326}
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureSpecConfig.TypeNamePath
+import org.locationtech.geomesa.utils.text.DateParsing
+
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+
+object GeoJsonInference {
+
+  val FieldsPath = "fields"
+  val OptionsPath = "options"
+
+  val attributeBuilder = new AttributeTypeBuilder()
+
+  val baseConfig: Config = ConfigFactory.empty()
+    .withValue("type", ConfigValueFactory.fromAnyRef("json"))
+    .withValue("feature-path", ConfigValueFactory.fromAnyRef("$[*]"))
+    .withValue("options", ConfigValueFactory.fromMap(Map(("line-mode", "multi"))))
+
+  def inferSft(filePath: String): SimpleFeatureType = {
+    val fileHandle: FileSystemDelegate.FileHandle = PathUtils.interpretPath(filePath).head
+    inferSft(fileHandle.open)
```

Comment:
probably makes sense to close this here, instead of in the method below. i.e.
`WithClose(fileHandle.open)(inferSft)`
As a general guideline, I try to make the `close` of a resource the responsibility of whatever opened it. Otherwise it's easy to lose track of things not being closed, or have them closed when you don't expect it.